realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1243,63 +1487,298 @@ public long count() {
     }
 
     /**
+     * Similar to {@link #findAllSorted(String[], boolean[])} but runs asynchronously
+     * from a worker thread
+     *
+     * @return immediately an empty {@link RealmResults}. Users need to register a listener
+     * io.realm.RealmResults#addChangeListener(RealmChangeListener) to be notified
+     * of the query completion.
+     * @throws java.lang.RuntimeException Any other error
+     * @see io.realm.RealmResults
+     */
+    public RealmResults<E> findAllSortedAsync(String fieldNames[], final boolean [] sortAscendings) {
+        if (fieldNames == null) {
+            throw new IllegalArgumentException("fieldNames cannot be 'null'.");
+        } else if (sortAscendings == null) {
+            throw new IllegalArgumentException("sortAscending cannot be 'null'.");
+        } else if (fieldNames.length == 0) {
+            throw new IllegalArgumentException("At least one field name must be specified.");
+        } else if (fieldNames.length != sortAscendings.length) {
+            throw new IllegalArgumentException(String.format("Number of field names (%d) and sort orders (%d) does not match.", fieldNames.length, sortAscendings.length));
+        }
+
+        if (fieldNames.length == 1 && sortAscendings.length == 1) {
+            return findAllSortedAsync(fieldNames[0], sortAscendings[0]);
+
+        } else {
+
+            // will use the Looper of the caller Realm to post the result
+            final WeakReference<Handler> weakHandler = new WeakReference<Handler>(realm.getHandler());
+
+            // Handover the query (to be used by a worker thread)
+            final long handoverQueryPointer = query.handoverQuery(realm.sharedGroup.getNativePointer());
+
+            // We need to use the same configuration to open a background SharedGroup to perform the query
+            final RealmConfiguration realmConfiguration = realm.getConfiguration();
+
+            final long indices[] = new long[fieldNames.length];
+            for (int i = 0; i < fieldNames.length; i++) {
```

Comment:
`findAllSortedAsync` uses directly the parameters as array compared to `findAllSorted` that uses a List then transform the list back to an array before calling when `tableView.sort(columnIndices, orders);` this should be optimised in a separate PR